src/actions/actions.ts

```diff
@@ -2971,6 +2974,93 @@ class MovementIBigWordTextObject extends MovementAWordTextObject {
 }
 
 @RegisterAction
+class SelectSentence extends TextObjectMovement {
+  keys = ["a", "s"];
+
+  public async execAction(position: Position, vimState: VimState): Promise<IMovement> {
+    let start: Position;
+    let stop: Position;
+
+    const currentSentenceBegin = position.getSentenceBegin({forward: false});
+    const currentSentenceNonWhitespaceEnd = currentSentenceBegin.getCurrentSentenceEnd();
+
+    if (currentSentenceNonWhitespaceEnd.isBefore(position)) {
+      // The cursor is on a trailing white space.
+      start = currentSentenceNonWhitespaceEnd.getRight();
+      stop = currentSentenceBegin.getSentenceBegin({forward: true}).getCurrentSentenceEnd();
+    } else {
+      const nextSentenceBegin = currentSentenceBegin.getSentenceBegin({forward: true});
+
+      // If the sentence has no trailing white spaces, `as` should include its leading white spaces.
+      if (nextSentenceBegin.isEqual(currentSentenceBegin.getCurrentSentenceEnd())) {
+        start = currentSentenceBegin.getSentenceBegin({forward: false}).getCurrentSentenceEnd().getRight();
+        stop = nextSentenceBegin;
+      } else {
+        start = currentSentenceBegin;
+        stop = nextSentenceBegin.getLeft();
+      }
+    }
+
+    if (vimState.currentMode === ModeName.Visual && !vimState.cursorPosition.isEqual(vimState.cursorStartPosition)) {
+      start = vimState.cursorStartPosition;
+
+      if (vimState.cursorPosition.isBefore(vimState.cursorStartPosition)) {
+        // If current cursor postion is before cursor start position, we are selecting sentences in reverser order.
+        if (currentSentenceNonWhitespaceEnd.isAfter(vimState.cursorPosition)) {
+          stop = currentSentenceBegin.getSentenceBegin({forward: false}).getCurrentSentenceEnd().getRight();
+        } else {
+          stop = currentSentenceBegin;
+        }
+      }
+    }
+
+    return {
+      start: start,
+      stop: stop
+    };
+  }
+}
+
+@RegisterAction
+class SelectInnerSentence extends TextObjectMovement {
+  keys = ["i", "s"];
+
+  public async execAction(position: Position, vimState: VimState): Promise<IMovement> {
+    let start: Position;
+    let stop: Position;
+
+    const currentSentenceBegin = position.getSentenceBegin({forward: false});
+    const currentSentenceNonWhitespaceEnd = currentSentenceBegin.getCurrentSentenceEnd();
+
+    if (currentSentenceNonWhitespaceEnd.isBefore(position)) {
+      // The cursor is on a trailing white space.
```

Comment:
The thing is Vim has a weird mechanism of handling leading or trailing spaces. The logic should be shared between words, sentences or paragraphs but since Vim is handling word, sentence and paragraph 's boundary differently, I still don't find a good way to reduce these redundant `if-else` statements. Will try later.